functional_tests/scylla_operator/test_functional.py

```diff
@@ -741,3 +746,46 @@ def _check_pod_stats(node):
     final_target_pod_stats = _check_pod_stats(target_node)
     assert final_target_pod_stats
     assert final_target_pod_stats.get("restart_count") == origin_target_pod_stats.get("restart_count")
+
+
+@pytest.mark.required_operator("v1.8.0")
+def test_ssl_autogenerated(db_cluster: ScyllaPodCluster, tmp_path: path.Path):
+
+    cluster_name = db_cluster.k8s_cluster.k8s_scylla_cluster_name
+
+    crt_filename = tmp_path / 'tls.crt'
+    key_filename = tmp_path / 'tls.key'
+    ca_filename = tmp_path / 'ca.crt'
+
+    crt_data = db_cluster.k8s_cluster.kubectl(fr"get secret/{cluster_name}-local-client-ca -o jsonpath='{{.data.tls\.crt}}'",
+                                              namespace=db_cluster.namespace)
+    key_data = db_cluster.k8s_cluster.kubectl(fr"get secret/{cluster_name}-local-client-ca -o jsonpath='{{.data.tls\.key}}'",
+                                              namespace=db_cluster.namespace)
+    ca_data = db_cluster.k8s_cluster.kubectl(fr"get secret/{cluster_name}-local-serving-ca -o jsonpath='{{.data.tls\.crt}}'",
+                                             namespace=db_cluster.namespace)
+
+    crt_filename.write_bytes(base64.decodebytes(bytes(crt_data.stdout.strip(), encoding='utf-8')))
+    key_filename.write_bytes(base64.decodebytes(bytes(key_data.stdout.strip(), encoding='utf-8')))
+    ca_filename.write_bytes(base64.decodebytes(bytes(ca_data.stdout.strip(), encoding='utf-8')))
+
+    log.debug(ca_filename)
+    log.debug(ca_filename.read_text())
+    log.debug(crt_filename)
+    log.debug(crt_filename.read_text())
+    log.debug(key_filename)
+    log.debug(key_filename.read_text())
+
+    cluster = Cluster(contact_points=[db_cluster.nodes[0].cql_ip_address], port=db_cluster.nodes[0].CQL_SSL_PORT)
+    ssl_context = ssl.SSLContext(protocol=ssl.PROTOCOL_SSLv23)
+    ssl_context.verify_mode = ssl.VerifyMode.CERT_REQUIRED  # pylint: disable=no-member
+
+    ssl_context.load_verify_locations(cadata=ca_filename.read_text())
+    ssl_context.load_cert_chain(keyfile=key_filename, certfile=crt_filename)
+
+    cluster.ssl_context = ssl_context
+
+    with cluster.connect(wait_for_all_pools=True) as session:
+        for host in cluster.metadata.all_hosts():
+            log.debug(host)
+        res = session.execute("SELECT * FROM system.local")
```

Comment:
System tables cannot be empty.
So, better to add an assertion to check that we have real data there.

Reply:
fixed